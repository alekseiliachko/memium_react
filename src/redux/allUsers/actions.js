import AccountOpenController from "../../api/AccountOpenController";

export const SET_USER_DATA = "SET_USER_DATA";
export const setUserData = (userId, userData) => ({
  type: SET_USER_DATA,
  userId,
  userData,
});

export const fetchUserData = (userId) => async (dispatch) => {
  const res = await AccountOpenController.getAccountsDetails(userId);
  dispatch(setUserData(userId, res.data));
};

export const SET_USER_AVATAR = "SET_USER_AVATAR";
export const setUserAvatar = (userId, avatar) => ({
  type: SET_USER_AVATAR,
  userId,
  avatar,
});
export const fetchUserAvatar = (userId) => async (dispatch) => {
  const res = await AccountOpenController.getAccountsAvatar(userId);
  dispatch(setUserAvatar(userId, res));
};
