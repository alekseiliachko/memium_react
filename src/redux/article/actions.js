import ArticlesController from "../../api/ArticlesController";

export const SET_ARTICLE_DATA = "SET_ARTICLE_DATA";

export const setArticleData = (articleId, articleData) => ({
  type: SET_ARTICLE_DATA,
  articleId,
  articleData,
});

export const fetchArticleData = (articleId) => async (dispatch) => {
  const res = await ArticlesController.getArticleById(articleId);
  dispatch(setArticleData(articleId, res.data));
};
