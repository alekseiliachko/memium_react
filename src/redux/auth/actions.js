import AuthController from "../../api/AuthController";

export const LOGIN_ATTEMPT = "LOGIN_ATTEMPT";
export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
export const LOG_OUT = "LOG_OUT";

export const SIGNUP_ATTEMPT = "SIGNUP_ATTEMPT";
export const SIGNUP_SUCCESS = "SIGNUP_SUCCESS";

export const AUTH_FAILURE = "AUTH_FAILURE";

export const setAuthStatusAttempt = () => ({
  type: LOGIN_ATTEMPT,
});

export const setAuthStatusSuccess = (username) => ({
  type: LOGIN_SUCCESS,
  payload: username,
});

export const logout = () => async (dispatch) => {
  localStorage.removeItem("memium_token");
  localStorage.removeItem("name");
  return {
    type: LOG_OUT,
  };
};

export const authFailure = (err) => ({
  type: AUTH_FAILURE,
  payload: err,
});

export const signupSuccess = () => ({
  type: SIGNUP_SUCCESS,
});

export const restoreSessionAttempt = () => (dispatch) => {
  dispatch(setAuthStatusAttempt());
  if (localStorage.memium_token) {
    AuthController.updateAuthHeader(localStorage.memium_token);
    dispatch(setAuthStatusSuccess(localStorage.name));
  } else {
    dispatch(logout());
  }
};

export const loginAttempt = (userInfo) => async (dispatch) => {
  dispatch(setAuthStatusAttempt());
  const res = await AuthController.login(userInfo);
  AuthController.updateAuthHeader(res.data.token);
  localStorage.setItem("name", res.data.username);
  dispatch(setAuthStatusSuccess(res.data.username));
};

export const signupAttempt = (userInfo) => async (dispatch) => {
  const res = AuthController.signup(userInfo);
  dispatch(signupSuccess());
};
