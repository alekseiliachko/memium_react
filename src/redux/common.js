export const LOADING_STATUS = {
  NOT_LOADED: "NOT_LOADED",
  LOADING: "LOADING",
  LOADED: "LOADED",
};
