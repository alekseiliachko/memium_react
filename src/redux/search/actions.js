import AccountsController from "../../api/AccountsController";

export const UPDATE_SEARCH_RESULTS = "UPDATE_SEARCH_RESULTS";
export const updateSearchResults = (articles, authors) => ({
  type: UPDATE_SEARCH_RESULTS,
  articles,
  authors,
});

export const fetchSearchResults = (search) => async (dispatch) => {
  const res = await AccountsController.getArticlesBySearch(search);
  dispatch(updateSearchResults(res.data.articles, res.data.accounts));
};
