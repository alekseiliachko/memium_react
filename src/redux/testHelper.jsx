import {Provider} from "react-redux";
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import React from "react";

export const getStubStoreData =  () => require('./testData.json');

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

export const withStore = (initialStore) => {
    const store = mockStore(initialStore || getStubStoreData());
    return function Wrapper({ children }) {
        return <Provider store={store}>
            {children}
        </Provider>
    }
}
