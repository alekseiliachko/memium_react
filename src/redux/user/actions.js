import AccountsController from "../../api/AccountsController";

export const AVATAR_LOADED = "AVATAR_LOADED";
export const ATTEMPT_LOAD_AVATAR = "ATTEMPT_LOAD_AVATAR";

export const setLoadedAvatar = (src) => ({
  payload: src,
  type: AVATAR_LOADED,
});

export const avatarLoadingAttempt = () => ({
  type: ATTEMPT_LOAD_AVATAR,
});

export const loadAvatar = () => async (dispatch) => {
  dispatch(avatarLoadingAttempt());
  const res = await AccountsController.getAvatarImage();
  dispatch(setLoadedAvatar(res.data));
};
