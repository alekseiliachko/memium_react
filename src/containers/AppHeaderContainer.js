import AppHeader from "../components/AppHeader";
import { connect } from "react-redux";
import { logout } from "../redux/auth/actions";

const mapStateToProps = (store) => ({
  username: store.authReducer.username,
});

const mapDispatchToProps = (dispatch) => ({
  onLogout: () => dispatch(logout()),
});

export default connect(mapStateToProps, mapDispatchToProps)(AppHeader);
