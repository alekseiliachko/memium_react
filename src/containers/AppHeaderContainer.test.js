import React from "react";
import { withStore } from "../redux/testHelper";
import { MemoryRouter } from "react-router";
import { createMount } from "@material-ui/core/test-utils";
import { AppHeaderContainer } from "./AppHeaderContainer";
import { loadAvatar } from "../redux/user/actions";
jest.mock("../redux/user/actions");

describe("appheader creation page", () => {
  it("renders correctly", async () => {
    const Comp = withStore();

    loadAvatar.mockReturnValue({ type: "who_cares" });

    const renderedValue = createMount()(
      <Comp>
        <MemoryRouter>
          <AppHeaderContainer />
        </MemoryRouter>
      </Comp>
    );

    expect(renderedValue.html()).toMatchSnapshot();
  });
});
