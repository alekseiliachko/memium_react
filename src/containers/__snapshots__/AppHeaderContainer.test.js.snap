// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appheader creation page renders correctly 1`] = `"<div><header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-header-5 MuiPaper-elevation4\\"><div class=\\"MuiContainer-root MuiContainer-maxWidthLg\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><a class=\\"makeStyles-link-2\\" href=\\"/home\\"><div class=\\"MuiBox-root MuiBox-root-7\\"><img src=\\"Logo.svg\\" alt=\\"memium logo\\"><h5 class=\\"MuiTypography-root makeStyles-title-4 MuiTypography-h5\\">Memium</h5></div></a><div class=\\"makeStyles-divider-3\\"></div><button class=\\"MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-1 MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"search\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><div><div class=\\"MuiAvatar-root MuiAvatar-circle makeStyles-avatar-6\\"><img src=\\"blob:http://localhost:3000/3dcb8dbb-23c7-4abf-8958-52663e0013da\\" class=\\"MuiAvatar-img\\"></div></div><div></div></div></div></header></div>"`;
